pygitguardian/client.py

```diff
@@ -347,3 +354,37 @@ def quota_overview(
         obj.status_code = resp.status_code
 
         return obj
+
+    # For IaC Scans
+    def iac_directory_scan(
+        self,
+        directory: bytes,
```

Comment:
The other endpoints take filenames as `str`. For consistency, it would be better if `directory` was `str` too (unless that causes an issue?).

Reply:
Oh I just realized `directory` is actually the bytes of a tar.gz of the directory. In this case it makes sense for it to be `bytes`, but the argument name is not adapted. It's tricky to find another name though 🤔. `tarred_directory_bytes`? Going to ask around.

Reply:
Alternatively, we could make `directory` be the path of the directory to scan, and make `iac_directory_scan()` create the tar bytes. After all, the fact that the API expects a tar archive can be considered to be an implementation detail.

Bonus point: this would let you pass the BytesIO instance directly in the `files` directory, I think.

Reply:
Thanks for the review! I've been thinking about I could transfer `create_tar` to `py-gitguardian` but I don't see an easy way. To do so, we would need to pass `files` as argument to `iac_directory_scan()` and it's a `Files` object so we would need to refactor everything related to `Files` won't we?

Reply:
Good point, we don't want to move `Files` to `py-gitguardian`.

Looking at the code, `create_tar()` only needs the paths. What we can do is make `iac_directory_scan()` take a `root_path: str` and a `files: List[str]` arguments, where `files` would be the list of paths to scan. ggshield can get the file list using the `Files.filenames` property (this badly named property returns the paths to the files).

Reply:
Thanks for the insight, it is indeed possible to do it the way you describe, however we would have the same issue with `Files` if we want to transfer the tests for `create_tar` in `tests/unit/iac/test_utils.py` I believe.

Reply:
Ah, what I had in mind is we would also move `create_tar` and its tests to pygitguardian, but a simplified version which takes a `root_path` and a list of `files: Iterable[str]`.

Thus `iac_directory_scan()` would look like this:

```python
    def iac_directory_scan(
        self,
        root_path: str,
        files: Iterable[str],
        scan_parameters: IaCScanParameters,
        extra_headers: Optional[Dict[str, str]] = None,
    ) -> Union[Detail, IaCScanResult]:

        directory = _create_tar(root_path, files)
        # ...
        try:
            resp = self.post(
                files={
                    "directory": directory,
                },
                # ...
            )
# ...

def _create_tar(root_path: str, files: Iterable[str]) -> bytes:
    # ...
```

Reply:
Thanks for the clarification, I've pushed a new commit that does this.